apps/src/gamelab/AnimationPicker/AnimationPickerBody.jsx

```diff
@@ -17,7 +20,18 @@ const AnimationPickerBody = React.createClass({
     onUploadClick: React.PropTypes.func.isRequired
   },
 
+  getInitialState() {
+    return {
+      searchQuery: ''
+    };
+  },
+
+  onSearchQueryChange(value) {
+    this.setState({searchQuery: value});
+  },
+
   render() {
+    var pageOfResults = searchAnimations(this.state.searchQuery);
```

Comment:
In this case, we are actually modifying the component's local `state` as the user types in the search box, which will not cause `componentWillReceiveProps` to occur.  Conversely, we could call this from `onSearchQueryChange` but that wouldn't handle the initial render case so we'd need to do it in `componentWillMount` too... in the end, I think it's best to trust React to only call `render()` when it's needed (it's usually good at that) and optimize later if it becomes an issue.  If you look at this component's props there's really nothing other than the search query that should trigger a re-render, so I suspect this will be fine.